diff --git a/src/printer.js b/src/printer.js
index 206b4fcad71822e5c92b36fb469346a4cdd56bfb..a08b7c0b0ac7ccd9cbf18f1a1affe62f48d15b16 100644
--- a/src/printer.js
+++ b/src/printer.js
@@ -1,13 +1,13 @@
 import EventEmitter from "events";
 import puppeteer from "puppeteer";
 
-import path from "path";
 import fs from "fs";
+import path from "path";
 import { fileURLToPath } from "url";
 
 import { PDFDocument } from "pdf-lib";
-import { setTrimBoxes, setMetadata } from "./postprocesser.js";
 import { parseOutline, setOutline } from "./outline.js";
+import { setMetadata, setTrimBoxes } from "./postprocesser.js";
 
 const currentPath = fileURLToPath(import.meta.url);
 const dir = process.cwd();
@@ -35,6 +35,7 @@ class Printer extends EventEmitter {
 		this.styles = options.styles || [];
 		this.enableWarnings = options.enableWarnings || false;
 		this.extraHTTPHeaders = options.extraHTTPHeaders || {};
+		this.extraCookies = options.extraCookies || [];
 
 		this.pages = [];
 
@@ -85,6 +86,8 @@ class Printer extends EventEmitter {
 
 			page.setExtraHTTPHeaders(this.extraHTTPHeaders);
 
+			page.setCookie(...this.extraCookies);
+
 			await page.emulateMediaType(this.emulateMedia);
 
 			if (this.overrideDefaultBackgroundColor) {
