diff --git a/src/printer.js b/src/printer.js
index 206b4fcad71822e5c92b36fb469346a4cdd56bfb..958bb64ef7d03e85b01a8c54c06a663b429c684a 100644
--- a/src/printer.js
+++ b/src/printer.js
@@ -1,13 +1,13 @@
 import EventEmitter from "events";
 import puppeteer from "puppeteer";
 
-import path from "path";
 import fs from "fs";
+import path from "path";
 import { fileURLToPath } from "url";
 
 import { PDFDocument } from "pdf-lib";
-import { setTrimBoxes, setMetadata } from "./postprocesser.js";
 import { parseOutline, setOutline } from "./outline.js";
+import { setMetadata, setTrimBoxes } from "./postprocesser.js";
 
 const currentPath = fileURLToPath(import.meta.url);
 const dir = process.cwd();
@@ -35,6 +35,7 @@ class Printer extends EventEmitter {
 		this.styles = options.styles || [];
 		this.enableWarnings = options.enableWarnings || false;
 		this.extraHTTPHeaders = options.extraHTTPHeaders || {};
+		this.extraCookies = options.extraCookies || [];
 
 		this.pages = [];
 
@@ -85,6 +86,8 @@ class Printer extends EventEmitter {
 
 			page.setExtraHTTPHeaders(this.extraHTTPHeaders);
 
+			page.setCookie(...this.extraCookies);
+
 			await page.emulateMediaType(this.emulateMedia);
 
 			if (this.overrideDefaultBackgroundColor) {
@@ -155,9 +158,12 @@ class Printer extends EventEmitter {
 						base.setAttribute("href", url);
 					}, url);
 				}
-
 			} else {
-				await page.goto(url);
+				const response = await page.goto(url);
+
+				if (!response || !response.ok()) {
+					throw new Error(`Failed to load page: ${response ? response.status() : 'No response'}`);
+				}
 			}
 
 			this.content = await page.content();
